cmake/modules/unittest.cmake

```diff
@@ -1,8 +1,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
 cmake_minimum_required(VERSION 3.20.0)
-cmake_policy(SET CMP0000 OLD)
-cmake_policy(SET CMP0002 NEW)
```

Comment:
commit message could mention why these are removed

Reply:
Done.

Reply:
Actually, I removed this change from this PR, and instead created a dedicated PR for this cleanup here: https://github.com/zephyrproject-rtos/zephyr/pull/41316